dlt/common/data_writers/buffered.py

```diff
@@ -148,8 +147,10 @@ def import_file(
         if with_extension:
             spec = self.writer_spec._replace(file_extension=with_extension)
         with self.alternative_spec(spec):
-            self._rotate_file()
+            self._rotate_file(is_imported_file=True)
```

Comment:
check wether the file is compressed before you rotate the file, set this new parameter mentioned below to true or false accordingly, then you do not have to mess with the filename at all here as it is set properly in the rotate_file method.